add a task about Kubernetes Services for egress

vadimeisenbergibm opened the pull request:
Replaces https://github.com/istio/istio.io/pull/3899

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

frankbu commented on content/docs/tasks/traffic-management/egress/egress-kubernetes-services/index.md:
```diff
@@ -0,0 +1,247 @@
+    metadata:
+      name: my-httpbin
+    spec:
+      host: my-httpbin.default.svc.cluster.local
```
Maybe it would be better to include the explicit `tlsSettings` with mode set to `DISABLE`. The docs aren't really clear what the default is, and I think there's been some discussion about possibly honoring the `tlsSettings` of any other existing `DestinationRule` on the lookup path if the selected `DestinationRule` has other things (like subsets) but no explicit `tlsSettings` defined.

vadimeisenbergibm replied:
Agree, good point! Handled in https://github.com/istio/istio.io/pull/4710/commits/5df8d209be2b4a05790bcee5cd23db071ccf3f9c

rcaballeromx commented on .spelling:
```diff
@@ -384,6 +384,7 @@ pluggable
 png
 pprof
 pre-connected
+pre-Istio
```
Please reword to avoid adding this entry to the spelling exceptions.

rcaballeromx commented on content/docs/tasks/traffic-management/egress/egress-kubernetes-services/index.md:
```diff
@@ -0,0 +1,253 @@
+weight: 60
+---
+
+Sometimes, for various reasons, for example in pre-Istio scenarios,
```
Please be clear and explicit about the scenarios where the k8s `ExternalName` or `Endpoints are used to access external services. As a reader, I need to be sure that this is the task I need before I begin going through the steps.

rcaballeromx commented:
I need to take a closer look at the content. I've added the the do-not-merge label for me to be able to complete the review.

rcaballeromx commented on content/docs/tasks/traffic-management/egress/egress-kubernetes-services/index.md:
```diff
@@ -0,0 +1,253 @@
+access external services.
+This task shows that these Kubernetes mechanisms for accessing external services continue to work with Istio.
+The only configuration step you must perform is to disable Istio's
+[mutual TLS](/docs/concepts/security/#mutual-tls-authentication).
```
Requiring folks to disable mTLS to get common K8s features is not the right approach, IMO. @geeknoid WDYT?
At the very least, readers will want to know why that is the case.

vadimeisenbergibm replied:
@rcaballeromx You are right, other TLS mode, for example SIMPLE, should be used instead of Istio mutual TLS.

frankbu replied:
I agree with @rcaballeromx, you should add a couple of sentences to explain what is needed and why.

vadimeisenbergibm commented:
@rcaballeromx @frankbu I addressed your comments, thanks a lot for your feedback.

I thought about the motivation for this task and it became clear for me that it is not related to pre-Istio scenarios. Kubernetes external services and endpoints provide location transparency and can be used with or without Istio. So this task is even more relevant than I thought initially.

I also addressed the TLS mode. For HTTP cases, I recommend to use Egress TLS origination, explained in a separate task. For HTTPS cases, the traffic is already encrypted by the workload and Istio should not perform additional encryption (otherwise the external service will not be able to decrypt the traffic twice and will fail).

frankbu commented:
/retest

geeknoid commented:
Could you update this PR? The files needs to move into the content/en directory.

Thanks.

istio-testing commented:
@vadimeisenbergibm: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio.io | 32ad2e41776f37937993cfb9eefde96bddf1fcee | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/4710/lint_istio.io/483) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

vadimeisenbergibm commented:
/test lint_istio.io

vadimeisenbergibm commented:
@geeknoid

>  Could you update this PR? The files needs to move into the content/en directory.

Done.

istio-testing commented:
@vadimeisenbergibm: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio.io | 32ad2e41776f37937993cfb9eefde96bddf1fcee | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/4710/lint_istio.io/490) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

frankbu commented:
/retest

istio-testing commented:
@vadimeisenbergibm: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio.io | 32ad2e41776f37937993cfb9eefde96bddf1fcee | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/4710/lint_istio.io/491) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

vadimeisenbergibm commented:
/test lint_istio.io

istio-testing commented:
In response to a cherrypick label: new pull request created: #5157
